File: jbake-core/src/main/java/org/jbake/template/ModelExtractor.java
package org.jbake.template;

import org.jbake.app.ContentStore;
import org.jbake.template.model.TemplateModel;


/**
 * @param <T> the type of data returned by this model extractor
 * @author ndx
 */
public interface ModelExtractor<T> {

    T get(ContentStore db, TemplateModel model, String key);

}

File: jbake-core/src/main/java/org/jbake/template/ModelExtractor.kt
package org.jbake.template

import org.jbake.app.ContentStore

/**
 * @param T  The type of data returned by this model extractor.
 * @author ndx
 */
interface ModelExtractor<T> {
    fun get(db: ContentStore, model: MutableMap<String, Any>, key: String): T?
}
